sr_object_symmetry/src/main.cpp

```diff
@@ -0,0 +1,318 @@
+                                   15, 125, 24, 1.0, 1.0, 1.0);
+              }
+            }
+            else if ((visState.cloudDisplay_ == VisState::REFLECTIONAL_SYMMETRIES) &&
+                     (symmetries_T.getReflectional().size() > 0))
+            {
+              std::vector<sym::ReflectionalSymmetry> symmetryDisplay;
+              std::vector<int> symmetryDisplayIds;
+              std::string text;
+              if (visState.cloudDisplay_ == VisState::REFLECTIONAL_SYMMETRIES)
+              {
+                symmetryDisplay = symmetries_T.getReflectional();
+                for (size_t symId = 0; symId < symmetries_T.getReflectional().size(); symId++)
+                  symmetryDisplayIds.push_back(symId);
+                text = "Reflectional symmetry ";
+              }
+              visState.segIterator_ = utl::clampValueCircular<int>(visState.segIterator_, 0,
+                                                                   symmetryDisplayIds.size() - 1);
+              int symId = symmetryDisplayIds[visState.segIterator_];
+              utl::showPointCloudColor<PointT>(visualizer, cloudHighRes, "cloud", visState.pointSize_);
+              if (visState.delete_)
+              {
+                visState.delete_ = false;
+                symmetries_T.removeCurrReflectional(symId);
+                if (symmetries_T.getReflectional().size() == 0)
+                  visState.cloudDisplay_ = VisState::CLOUD;
+                visState.updateDisplay_ = true;
+              }
+              // Show symmetry
+              if (symmetries_T.getReflectional().size() > 0)
+              {
+                if (visState.showSymmetry_)
+                  sym::showReflectionalSymmetry(visualizer, symmetryDisplay[symId], "symmetry", 1);
+                text = "cloudInlierScores Score: " + std::to_string(symmetries_T.getReflectionalScores()[symId]);
+                visualizer.addText("Numpad 0: Delete current symmetry", 0, 0, 24, 1.0, 1.0, 1.0);
+                visualizer.addText("Numpad 1: Show Point Cloud", 0, 20, 24, 1.0, 1.0, 1.0);
+                visualizer.addText("Numpad 3: Show Rotational Symmetries", 0, 40, 24, 1.0, 1.0, 1.0);
+                visualizer.addText("Numpad 7: Show Reflectional Symmetries", 0, 60, 24, 1.0, 1.0, 1.0);
+                visualizer.addText("Q: Go to next object", 0, 80, 24, 1.0, 1.0, 1.0);
+                visualizer.addText(text, 0, 100, 24, 1.0, 1.0, 1.0);
+                visualizer.addText("Symmetry index ID" + std::to_string(visState.segIterator_) + " Size " +
+                                       std::to_string(symmetryDisplayIds.size()),
+                                   15, 125, 24, 1.0, 1.0, 1.0);
+              }
+            }
+          }
+          // Spin once
+          visualizer.spinOnce();
+          boost::this_thread::sleep(boost::posix_time::milliseconds(10));
+        }
+      }
+      YAML::Emitter yamlOut;
```

Comment:
Certain things, like this yaml output section, could be separate functions, to make the main more readable.
But no need to change it now.